### PR TITLE
dde-daemon: do not create /usr/share/uadp

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wineee** [comment]
It seems that only deepin-face(Face recognition service) calls this interface, no plans to port this:
https://github.com/linuxdeepin/deepin-face/commit/f187d4e75d7c4cf14f567bfb2eb7c5d97d81f2b8

I think we can delete this part of code directly as archlinux did
https://github.com/linuxdeepin/dde-daemon/blob/662e0f184d982ce2e3c4b76135a9fd4297102154/archlinux/PKGBUILD#L35-L36

File: pkgs/desktops/deepin/go-package/dde-daemon/default.nix [hunk 1]
```diff
@@ -67,6 +67,9 @@ buildGoPackage rec {
     substituteInPlace inputdevices/layout_list.go \
       --replace "/usr/share/X11/xkb" "${xkeyboard_config}/share/X11/xkb"
 
+    substituteInPlace system/uadp/crypto.go \
+      --replace "/usr/share/uadp" "/run/current-system/sw/share/uadp"
```
Comment (wineee) [inline pkgs/desktops/deepin/go-package/dde-daemon/default.nix:71]:
```suggestion
      --replace "/usr/share/uadp" "/var/lib/dde-daemon/uadp"
```
The /usr/share hierarchy is for  ***read-only*** architecture independent data files,  dde-daemon does not strictly comply with FHS


https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html
https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s08.html

**Review by wineee — CHANGES_REQUESTED** [review]

**Review by wineee — APPROVED** [review]

**Comment by wineee** [comment]
@ofborg build deepin.dde-daemon

**Comment by wegank** [comment]
Capture d’écran 2023-05-05 à 17 06 08

**Pull request merged**